internal/ethapi/api.go

```diff
@@ -1417,7 +1417,11 @@ func AccessList(ctx context.Context, b Backend, blockNrOrHash rpc.BlockNumberOrH
 		}
 		// Copy the original db so we don't modify it
 		statedb := db.Copy()
-		msg := types.NewMessage(args.from(), args.To, uint64(*args.Nonce), args.Value.ToInt(), uint64(*args.Gas), args.GasPrice.ToInt(), big.NewInt(0), big.NewInt(0), args.data(), accessList, false)
+		gasPrice := args.GasPrice.ToInt()
+		if gasPrice == nil {
+			gasPrice = new(big.Int)
+		}
+		msg := types.NewMessage(args.from(), args.To, uint64(*args.Nonce), args.Value.ToInt(), uint64(*args.Gas), gasPrice, big.NewInt(0), big.NewInt(0), args.data(), accessList, false)
```

Comment:
Why don't you use the `args.toMessage()` function instead? Perhaps you can re-assign the accessList after converting it to the message. So that you don't need to handle the gasPrice.

Reply:
Yep using to message now, and set the accesslist beforehand